From 90b3db59098e71ce23c553a2311acc366f8ac8a4 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 25 Jul 2013 15:43:30 +0800
Subject: [PATCH 219/310] ENGR00268883 wifi: refactor iprenew for p2p

add iprenew_p2p for all imx6 and imx5 platform.
In long time wifi display test, p2p will renew ip depending this svc.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 arm2_6dq/init.rc            |  5 -----
 arm2_6sl/init.rc            |  5 -----
 evk_6sl/init.rc             |  5 -----
 imx53_smd/init.freescale.rc |  5 -----
 imx5x/init.rc               | 10 ++++++++++
 imx6/etc/init.rc            | 10 ++++++++++
 sabreauto_6q/init.rc        |  5 -----
 sabresd_6dq/init.rc         |  5 -----
 8 files changed, 20 insertions(+), 30 deletions(-)

diff --git a/arm2_6dq/init.rc b/arm2_6dq/init.rc
index e54a593..e267994 100755
--- a/arm2_6dq/init.rc
+++ b/arm2_6dq/init.rc
@@ -71,11 +71,6 @@ service wpa_supplicant /system/bin/wpa_supplicant \
     disabled
     oneshot
 
-service iprenew_wlan0 /system/bin/dhcpcd -n
-    class late_start
-    disabled
-    oneshot
-
 on fs
 # mount ext4 partitions
     mount ext4 /dev/block/mmcblk0p5 /system
diff --git a/arm2_6sl/init.rc b/arm2_6sl/init.rc
index 97ac32f..646105f 100755
--- a/arm2_6sl/init.rc
+++ b/arm2_6sl/init.rc
@@ -64,11 +64,6 @@ service wpa_supplicant /system/bin/wpa_supplicant \
     disabled
     oneshot
 
-service iprenew_wlan0 /system/bin/dhcpcd -n
-    class late_start
-    disabled
-    oneshot
-
 on fs
 # mount ext4 partitions
     mount ext4 /dev/block/mmcblk0p5 /system
diff --git a/evk_6sl/init.rc b/evk_6sl/init.rc
index 8ecdef4..db0b38d 100644
--- a/evk_6sl/init.rc
+++ b/evk_6sl/init.rc
@@ -111,11 +111,6 @@ service wpa_supplicant /system/bin/wpa_supplicant -iwlan0 -Dnl80211
     disabled
     oneshot
 
-service iprenew_wlan0 /system/bin/dhcpcd -n
-    class main
-    disabled
-    oneshot
-
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
diff --git a/imx53_smd/init.freescale.rc b/imx53_smd/init.freescale.rc
index c9e62e1..0bc9475 100755
--- a/imx53_smd/init.freescale.rc
+++ b/imx53_smd/init.freescale.rc
@@ -113,10 +113,5 @@ service ing /system/bin/ingsvcd -c /system/etc/gps/Orion.ini
     group gps
     oneshot
 
-service iprenew_wlan0 /system/bin/dhcpcd -n
-    class late_start
-    disabled
-    oneshot
-
 on fs
     mount_all /fstab.freescale
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 4f44c91..9f0628f 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -612,6 +612,16 @@ service wifi_mac /system/bin/sh /system/etc/check_wifi_mac.sh
     class late_start
     oneshot
 
+service iprenew_wlan0 /system/bin/dhcpcd -n
+    class main
+    disabled
+    oneshot
+
+service iprenew_p2p /system/bin/dhcpcd -n
+    class main
+    disabled
+    oneshot
+
 service sshd /system/bin/start-ssh
     class main
     disabled
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 7e332e3..90019cd 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -657,6 +657,16 @@ service wifi_mac /system/bin/sh /system/etc/check_wifi_mac.sh
     class late_start
     oneshot
 
+service iprenew_wlan0 /system/bin/dhcpcd -n
+    class main
+    disabled
+    oneshot
+
+service iprenew_p2p /system/bin/dhcpcd -n
+    class main
+    disabled
+    oneshot
+
 # virtual sdcard daemon running as media_rw (1023)
 # default 4 threads have better performance to 2 threads.
 service sdcard /system/bin/sdcard -t4 /data/media /mnt/shell/emulated 1023 1023
diff --git a/sabreauto_6q/init.rc b/sabreauto_6q/init.rc
index f6d4368..fdf7065 100644
--- a/sabreauto_6q/init.rc
+++ b/sabreauto_6q/init.rc
@@ -139,11 +139,6 @@ service ing /system/bin/ingsvcd -c /system/etc/gps/Orion.ini
     group gps
     oneshot
 
-service iprenew_wlan0 /system/bin/dhcpcd -n
-    class main
-    disabled
-    oneshot
-
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
diff --git a/sabresd_6dq/init.rc b/sabresd_6dq/init.rc
index 67135bd..de323c6 100755
--- a/sabresd_6dq/init.rc
+++ b/sabresd_6dq/init.rc
@@ -143,11 +143,6 @@ service ing /system/bin/ingsvcd -c /system/etc/gps/Orion.ini
     group gps
     oneshot
 
-service iprenew_wlan0 /system/bin/dhcpcd -n
-    class main
-    disabled
-    oneshot
-
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
-- 
1.8.0

